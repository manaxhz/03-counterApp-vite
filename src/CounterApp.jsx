import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const CounterApp = ({value}) => {


  const [ counter, setCounter] = useState(value);

  const handleAdd = () => {
    // setCounter( counter + 1);
    setCounter( (c) => c + 1);

  }

  const handleReset = () => {
    // setCounter( counter + 1);
    setCounter( (c) => value);

  }

  const handleSubtract = () => {
    // setCounter( counter + 1);
    setCounter( (c) => c - 1);

  }



  return (
  <div> 
    <h1>CounterApp</h1>
    <h2> { counter } </h2>

  <div className=' space-x-1'>

      <button 
        onClick={handleSubtract}
        className='bg-transparent hover:bg-blue-500 text-blue-700 font-semibold hover:text-white py-2 px-4 border border-blue-500 hover:border-transparent rounded'>
          -1
      </button>

      <button 
        onClick={handleReset}
        className='bg-transparent hover:bg-blue-500 text-blue-700 font-semibold hover:text-white py-2 px-4 border border-blue-500 hover:border-transparent rounded'>
          reset
      </button>

      <button 
        onClick={handleAdd}
        className='bg-transparent hover:bg-blue-500 text-blue-700 font-semibold hover:text-white py-2 px-4 border border-blue-500 hover:border-transparent rounded'>
          +1
      </button>
  </div>
  </div>
  )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}
CounterApp.defaultProps = {
    value: 'No hay numeros',
}