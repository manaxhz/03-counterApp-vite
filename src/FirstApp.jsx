import PropTypes from 'prop-types';



export const FirstApp = ( {title, subTitle, name}) => {
 
  return (
    <div className="">
        {/* <code>{ JSON.stringify(newMessage)}</code> */}
        <h1 data-testid="test-title">{title}</h1>
        <p>{subTitle}</p>
        <p>{subTitle}</p>
        <p>{name}</p>

    </div >
  )
}


FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired, 
}

FirstApp.defaultProps = {
  name: 'Fernando Herrera',
  subTitle: 'No hay subtitlo',
  // title: 'No hay titulo',
}