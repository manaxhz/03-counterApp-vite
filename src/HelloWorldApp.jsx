// export const App = () => {
//   return (
//     <div><h1>Hola Mundo!!!!</h1></div>
//   )
// }

// export default App;


export const HelloWorldApp = () => {
  return (
    <h1>HelloWorldApp</h1>
  )
}
