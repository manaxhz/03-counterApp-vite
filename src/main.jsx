import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
// import { FirstApp } from './FirstApp';
// import { HelloWorldApp } from './HelloWorldApp';


import './index.css';

ReactDOM.createRoot( document.getElementById('root')).render(
    <React.StrictMode>

        {/* <FirstApp title="Hola, soy Vegeta" /> */}
        <CounterApp value={123} />
    </React.StrictMode>
)