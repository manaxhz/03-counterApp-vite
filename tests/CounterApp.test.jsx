import {fireEvent, render, screen } from "@testing-library/react"
import { CounterApp } from "../src/CounterApp"

describe('Pruebas en <CounterApp />', () => { 

    const initialValue= 100;

    test('debe de hacer match con el snapshoot', () => { 
        
        const {container} = render(<CounterApp />)

        expect(container).toMatchSnapshot();


     })

     test('debe de mostrar el valor inicial de 100', () => { 

        render(<CounterApp />)

        expect( screen.getByText(initialValue)).toBeTruthy();


      })

    test('debe de incrementar con el boton +1', () => { 

        render(<CounterApp />)
        fireEvent.click(screen.getByText('+1') )
        expect( screen.getByText('101')).toBeTruthy();

     })

     test('debe de decrementar con el boton -1', () => { 

        render(<CounterApp />)
        fireEvent.click(screen.getByText('-1') )
        // screen.debug()
        expect( screen.getByText('99')).toBeTruthy();

     })
     
     test('debe de funcionar el boton de reset', () => { 

        render(<CounterApp />)
        fireEvent.click(screen.getByText('-1') )
        fireEvent.click(screen.getByText('-1') )
        fireEvent.click(screen.getByText('-1') )

        // fireEvent.click(screen.getByText('reset') )
        // screen.debug()

        fireEvent.click( screen.getByRole('button', { name: 'btn-reset'}) )


        expect( screen.getByText('100')).toBeTruthy();

     })




 })