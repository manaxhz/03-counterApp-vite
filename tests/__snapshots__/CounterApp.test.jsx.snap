// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> debe de hacer match con el snapshoot 1`] = `
<div>
  <div>
    <h1>
      CounterApp
    </h1>
    <h2>
       
      100
       
    </h2>
    <div
      class=" space-x-1"
    >
      <button
        class="bg-transparent hover:bg-blue-500 text-blue-700 font-semibold hover:text-white py-2 px-4 border border-blue-500 hover:border-transparent rounded"
      >
        -1
      </button>
      <button
        aria-label="btn-reset"
        class="bg-transparent hover:bg-blue-500 text-blue-700 font-semibold hover:text-white py-2 px-4 border border-blue-500 hover:border-transparent rounded"
      >
        reset
      </button>
      <button
        class="bg-transparent hover:bg-blue-500 text-blue-700 font-semibold hover:text-white py-2 px-4 border border-blue-500 hover:border-transparent rounded"
      >
        +1
      </button>
    </div>
  </div>
</div>
`;
